third_party/move/move-prover/boogie-backend/src/bytecode_translator.rs

```diff
@@ -1492,6 +1737,36 @@ impl<'env> FunctionTranslator<'env> {
                             field_sel,
                         );
                     },
+                    BorrowVariantField(mid, sid, variants, inst, field_offset) => {
+                        let inst = &self.inst_slice(inst);
+                        let src_str = str_local(srcs[0]);
+                        let deref_src_str = format!("$Dereference({})", src_str);
+                        let dest_str = str_local(dests[0]);
+                        let struct_env = env.get_module(*mid).into_struct(*sid);
+                        self.check_intrinsic_select(attr_id, &struct_env);
+                        emit!(writer, "if (");
+                        let mut select_condition = vec![];
+                        for variant in variants {
+                            let struct_variant_name =
+                                boogie_struct_variant_name(&struct_env, inst, *variant);
+                            let call = format!("{} is {}", deref_src_str, struct_variant_name);
+                            select_condition.push(call.clone());
+                        }
+                        let field_env = struct_env
+                            .get_field_by_offset_optional_variant(Some(variants[0]), *field_offset);
+                        let field_sel = boogie_field_sel(&field_env);
+                        emitln!(writer, "{}) {{", select_condition.join(" || "));
+                        emitln!(
+                            writer,
+                            "{} := $ChildMutation({}, {}, {}->{});",
+                            dest_str,
+                            src_str,
+                            field_offset,
+                            deref_src_str,
+                            field_sel,
+                        );
+                        emitln!(writer, "} else { call $Abort(4038); }");
```

Comment:
Where do you got the 4038 from? This should be `$ExecFailureAbort` for VM Status failure.

Reply:
done